tests/baselines/reference/trackedSymbolsNoCrash.js

```diff
@@ -743,204 +743,6 @@ export interface Node99 {
-}> | Extract<ast.Node42, {
-    kind: NodeType;
-}> | Extract<ast.Node43, {
-    kind: NodeType;
-}> | Extract<ast.Node44, {
-    kind: NodeType;
-}> | Extract<ast.Node45, {
-    kind: NodeType;
-}> | Extract<ast.Node46, {
-    kind: NodeType;
-}> | Extract<ast.Node47, {
-    kind: NodeType;
-}> | Extract<ast.Node48, {
-    kind: NodeType;
-}> | Extract<ast.Node49, {
-    kind: NodeType;
-}> | Extract<ast.Node50, {
-    kind: NodeType;
-}> | Extract<ast.Node51, {
-    kind: NodeType;
-}> | Extract<ast.Node52, {
-    kind: NodeType;
-}> | Extract<ast.Node53, {
-    kind: NodeType;
-}> | Extract<ast.Node54, {
-    kind: NodeType;
-}> | Extract<ast.Node55, {
-    kind: NodeType;
-}> | Extract<ast.Node56, {
-    kind: NodeType;
-}> | Extract<ast.Node57, {
-    kind: NodeType;
-}> | Extract<ast.Node58, {
-    kind: NodeType;
-}> | Extract<ast.Node59, {
-    kind: NodeType;
-}> | Extract<ast.Node60, {
-    kind: NodeType;
-}> | Extract<ast.Node61, {
-    kind: NodeType;
-}> | Extract<ast.Node62, {
-    kind: NodeType;
-}> | Extract<ast.Node63, {
-    kind: NodeType;
-}> | Extract<ast.Node64, {
-    kind: NodeType;
-}> | Extract<ast.Node65, {
-    kind: NodeType;
-}> | Extract<ast.Node66, {
-    kind: NodeType;
-}> | Extract<ast.Node67, {
-    kind: NodeType;
-}> | Extract<ast.Node68, {
-    kind: NodeType;
-}> | Extract<ast.Node69, {
-    kind: NodeType;
-}> | Extract<ast.Node70, {
-    kind: NodeType;
-}> | Extract<ast.Node71, {
-    kind: NodeType;
-}> | Extract<ast.Node72, {
-    kind: NodeType;
-}> | Extract<ast.Node73, {
-    kind: NodeType;
-}> | Extract<ast.Node74, {
-    kind: NodeType;
-}> | Extract<ast.Node75, {
-    kind: NodeType;
-}> | Extract<ast.Node76, {
-    kind: NodeType;
-}> | Extract<ast.Node77, {
-    kind: NodeType;
-}> | Extract<ast.Node78, {
-    kind: NodeType;
-}> | Extract<ast.Node79, {
-    kind: NodeType;
-}> | Extract<ast.Node80, {
-    kind: NodeType;
-}> | Extract<ast.Node81, {
-    kind: NodeType;
-}> | Extract<ast.Node82, {
-    kind: NodeType;
-}> | Extract<ast.Node83, {
-    kind: NodeType;
-}> | Extract<ast.Node84, {
-    kind: NodeType;
-}> | Extract<ast.Node85, {
-    kind: NodeType;
-}> | Extract<ast.Node86, {
-    kind: NodeType;
-}> | Extract<ast.Node87, {
-    kind: NodeType;
-}> | Extract<ast.Node88, {
-    kind: NodeType;
-}> | Extract<ast.Node89, {
-    kind: NodeType;
-}> | Extract<ast.Node90, {
-    kind: NodeType;
-}> | Extract<ast.Node91, {
-    kind: NodeType;
-}> | Extract<ast.Node92, {
-    kind: NodeType;
-}> | Extract<ast.Node93, {
-    kind: NodeType;
-}> | Extract<ast.Node94, {
-    kind: NodeType;
-}> | Extract<ast.Node95, {
-    kind: NodeType;
-}> | Extract<ast.Node96, {
-    kind: NodeType;
-}> | Extract<ast.Node97, {
-    kind: NodeType;
-}> | Extract<ast.Node98, {
-    kind: NodeType;
-}> | Extract<ast.Node99, {
+export declare const isNodeOfType: <NodeType extends ast.SyntaxKind>(nodeType: NodeType) => (node: ast.Node | null | undefined) => node is Extract<ast.Node, {

```

Comment:
This change is probably the biggest "these kinds of changes [to reuse more input nodes] are obviously improvements for everybody" demonstrator in our test suite.

Reply:
Very nice, though I wonder if the crash from this test case was related to the printing or not...